banks-client/src/error.rs

```diff
@@ -19,6 +19,13 @@ pub enum BanksClientError {
 
     #[error("transport transaction error: {0}")]
     TransactionError(#[from] TransactionError),
+
+    #[error("simulation error: {err:?}, logs: {logs:?}, units_consumed: {units_consumed:?}")]
+    SimulationError {
+        err: TransactionError,
+        logs: Vec<String>,
+        units_consumed: u64,
+    },
 }
 
 impl BanksClientError {
```

Comment:
What do you think about `unwrap()` also pulling out a `SimulationError`?

Reply:
Mm, so this?
```
pub fn unwrap(&self) -> Box<dyn std::error::Error>
```
We'll have to make sure that doesn't break anything already expecting an explicit TransactionError.

Or wait, did you mean `unwrap` handling the BanksClientError::SimulationError to return the TransactionError piece?

Reply:
> did you mean unwrap handling the BanksClientError::SimulationError to return the TransactionError piece?

I was thinking more this one, but happy to be overruled! Otherwise, tests that just do `unwrap()` right now, and upgrade to doing a simulation, will have to do some sort of behavior to extract the TransactionError.  But maybe we just assume that those tests will stick with *not* performing a simulation

Reply:
Oh excellent, I'm much happier with that approach; seems like a no-brainer. And it seems fair to ask the user to handle the BanksClientError specifically, if they need the logs or units_consumed.